model/task_queue_service.go

```diff
@@ -231,6 +246,14 @@ func (t *basicCachedDispatcherImpl) FindNextTask(spec TaskSpec) *TaskQueueItem {
 	// If units (map[string]schedulableUnit) is empty, the queue has been emptied. Reset order as an optimization so that the
 	// service no longer needs to iterate over it and check each item against the map.
 	if len(t.units) == 0 {
+		grip.Debug(message.Fields{
+			"ticket":       "EVG-6289",
+			"message":      "t.units (map[string]schedulableUnit) is empty, resetting t.order = []string{} ",
+			"order_length": len(t.order),
+			"units_length": len(t.units),
+			"is_mismatch":  !(len(t.order) == len(t.units)),
```

Comment:
This is fine to leave in, but it's is unlikely to matter, since we only clear the order when the units map is empty.

Reply:
OK.